cdap-proto/src/main/java/co/cask/cdap/proto/NamespaceConfig.java

```diff
@@ -28,14 +29,42 @@
   @SerializedName("scheduler.queue.name")
   private final String schedulerQueueName;
 
-  public NamespaceConfig(String schedulerQueueName) {
+  @SerializedName("hdfs.directory")
+  private final String hdfsDirectory;
+
+  @SerializedName("hbase.namespace")
+  private final String hbaseNamespace;
+
+  @SerializedName("hive.database")
+  private final String hiveDatabase;
+
+  public NamespaceConfig(@Nullable String schedulerQueueName, @Nullable String hdfsDirectory, @Nullable String
```

Comment:
nit: fix indentation by moving the entire hbaseNamespace declaration to the next line